### ISSUE TITLE
Summaries don't get updated for all rooms when enabling session backup and refreshed since initial sync

### ISSUE THREAD

**Issue by bwindels** [element-hq/hydrogen-web#109]
Since: 0.1.0

Repro steps:

 1. log in
 1. refresh
 1. open account
 1. go to an e2ee room (that can't be decrypted)
 1. enable session backup
 1. e2ee rooms are still not sorted in the room list according to their latest activity

**Comment by bwindels** [comment]
Furthermore, it seems like the room list isn't updated at all (even for non-e2ee rooms). At least unread isn't set.
When clicking rooms that should have updated, they jumped to their correct position though, so feels like a missing emit.
Also, when reloading the app and trying to open the session again, I got `Something went wrong: Cannot read property 'eventIndex' of undefined.` for room `https://matrix.to/#/!mjbDjyNsRXndKLkHIe:matrix.org` on this line in the sync writer (happened once earlier with the same room id, I fixed by logging in again):
```
this._lastLiveKey = new EventKey(liveFragment.id, lastEvent.eventIndex);
```
with lastevent being undefined. Looking in the database, there are indeed no events for that room.

**Comment by bwindels** [comment]
The session definitely managed to load earlier, after having refreshed after initial sync. so something must have removed the events??

**Comment by bwindels** [comment]
There seem to be 3 sides to this bug (or even 3 bugs): 
 - the room summaries of e2ee rooms don't get updated when enabling session backup after having refreshed since you did initial sync.
 - the room summaries don't update at all, not even for non-e2ee rooms
 - all events in the Synapse Admins room are removed from the timeline store, making it crash on login. Note that there is still a fragment for this room in the database!

**Comment by bwindels** [comment]
Ok, can repro it now on android as well: 
 - room summaries don't update for any rooms
 - for non-e2ee rooms, they are updated in the list when you click them (so an emit is missing/not working)
 - for e2ee rooms, they just don't update in the room list when receiving messages (that can be decrypted), not even after clicking/opening them.

On android, I don't have the `Something went wrong: Cannot read property 'eventIndex' of undefined.` issue though, so that might be a different issue?

**Comment by bwindels** [comment]
This should be fixed in 46abafdccca85abb28b19ae2058ac7f92aed4706 but unsure if this fixes `Cannot read property 'eventIndex' of undefined` as I still don't have any idea how events got removed after they were synced...

**Comment by bwindels** [comment]
`Cannot read property 'eventIndex' of undefined` was split out in a different issue. The bug where the room summary wasn't updated at all was also fixed, see above commit link (I didn't create a separate ticket for this). The original bug is still valid though, but is better described at #111, so closing this one.
